### ISSUE TITLE
Add support for search query params to /latest endpoints

### ISSUE THREAD

**Issue by matchai** [axllent/mailpit#206]
When testing with emails, I'll often want to pull up the latest email sent by my test runner, but with tests running in parallel, `/latest.html` can bring about race conditions, showing the wrong email.

By allowing `/latest` endpoints to support search-like query params, I can avoid these race conditions by filtering by the sender or recipient, which can be unique per test.

**Comment by axllent** [comment]
Hi @matchai. I understand the issue you are having with a race condition. You can actually do this already, although it uses two API calls:

1. `/api/v1/search?query=<query>&limit=1` would return the latest [message summary](https://mailpit.axllent.org/docs/api-v1/view.html#get-/api/v1/search) (JSON array) matching your query. This summary would include the database `ID` of the message.
2. Using the `ID`, you could then fetch `/view/<ID>.html`

Does this not do what you are trying to achieve?

**Comment by matchai** [comment]
That totally works though, as a presumably common use-case, it would be great to have a convenient way to do it at once.
No worries though if you consider it to be out of scope for the project ☺️

**Comment by axllent** [comment]
@matchai I have just added this feature in [v1.10.0](https://github.com/axllent/mailpit/releases/tag/v1.10.0). You can find more info on https://mailpit.axllent.org/docs/integration/. Please confirm this does what you expect?

**Comment by axllent** [comment]
I'm closing this issue now as support for optional query parameters in `/view/latest.html?query=...` and `/view/latest.txt?query=...` was released a week ago in the previous version on Mailpit. Please feel free to reopen this issue if you experience any issues.
